indico/modules/events/registration/util.py

```diff
@@ -995,3 +997,22 @@ def get_persons(registrations, include_accompanying_persons=False):
                             'registration': registration,
                             'is_accompanying': True} for person in registration.accompanying_persons)
     return persons
+
+
+def resize_uploaded_registration_picture(source, size):
+    """Resize uploaded registration picture to the given size."""
+    try:
+        picture = Image.open(source)
+        picture = ImageOps.exif_transpose(picture)
+        size_x, size_y = picture.size
+        if max(size_x, size_y) < size:
+            source.seek(0)
+            return source
+        ratio = float(size) / max(size_x, size_y)
+        picture = picture.resize((int(ratio * size_x), int(ratio * size_y)), Image.Resampling.BICUBIC)
+        image_bytes = BytesIO()
+        picture.save(image_bytes, 'PNG')
```

Comment:
Wouldn't a jpg image be better here? People will typically upload a photo, and jpeg gives you much better compression for that kind of image